addons/sale_loyalty/models/sale_order.py

```diff
@@ -1231,3 +1232,22 @@ def _try_apply_code(self, code):
                 return apply_result
             coupon = apply_result.get('coupon', self.env['loyalty.card'])
         return self._get_claimable_rewards(forced_coupons=coupon)
+
+    def _validate_order(self):
+        """
+        Confirm the sale order and handle zero-amount invoicing.
+
+        If the order total is zero and automatic invoicing is enabled, it creates and posts an invoice.
```

Comment:
`Override of sale to create invoice for zero amount order`
Seems better you can add some more thing one top of that